test/base/resources/eventing.go

```diff
@@ -0,0 +1,190 @@
+}
+
+// Channel returns a Channel with the specified provisioner.
+func Channel(name, provisioner string) *eventingv1alpha1.Channel {
+	return &eventingv1alpha1.Channel{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: eventingv1alpha1.ChannelSpec{
+			Provisioner: clusterChannelProvisioner(provisioner),
+		},
+	}
+}
+
+// WithSubscriberForSubscription returns an option that adds a Subscriber for the given Subscription.
+func WithSubscriberForSubscription(name string) SubscriptionOption {
+	return func(s *eventingv1alpha1.Subscription) {
+		if name != "" {
+			s.Spec.Subscriber = &eventingv1alpha1.SubscriberSpec{
+				Ref: pkgTest.CoreV1ObjectReference("Service", "v1", name),
+			}
+		}
+	}
+}
+
+// WithReply returns an options that adds a ReplyStrategy for the given Subscription.
+func WithReply(name string, typemeta *metav1.TypeMeta) SubscriptionOption {
+	return func(s *eventingv1alpha1.Subscription) {
+		if name != "" {
+			s.Spec.Reply = &eventingv1alpha1.ReplyStrategy{
+				Channel: pkgTest.CoreV1ObjectReference(typemeta.Kind, typemeta.APIVersion, name),
+			}
+		}
+	}
+}
+
+// Subscription returns a Subscription.
+func Subscription(
+	name, channelName string,
+	channelTypeMeta *metav1.TypeMeta,
+	options ...SubscriptionOption,
+) *eventingv1alpha1.Subscription {
+	subscription := &eventingv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: eventingv1alpha1.SubscriptionSpec{
+			Channel: *channelRef(channelName, channelTypeMeta),
+		},
+	}
+	for _, option := range options {
+		option(subscription)
+	}
+	return subscription
+}
+
+// Broker returns a Broker.
+func Broker(name, provisioner string) *eventingv1alpha1.Broker {
+	return &eventingv1alpha1.Broker{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: eventingv1alpha1.BrokerSpec{
+			ChannelTemplate: &eventingv1alpha1.ChannelSpec{
+				Provisioner: clusterChannelProvisioner(provisioner),
+			},
+		},
+	}
+}
+
+// WithTriggerFilter returns an option that adds a TriggerFilter for the given Trigger.
+func WithTriggerFilter(eventSource, eventType string) TriggerOption {
+	return func(t *eventingv1alpha1.Trigger) {
+		triggerFilter := &eventingv1alpha1.TriggerFilter{
+			SourceAndType: &eventingv1alpha1.TriggerFilterSourceAndType{
+				Type:   eventType,
+				Source: eventSource,
+			},
+		}
+		t.Spec.Filter = triggerFilter
+	}
+}
+
+// WithBroker returns an option that adds a Broker for the given Trigger.
+func WithBroker(brokerName string) TriggerOption {
+	return func(t *eventingv1alpha1.Trigger) {
+		t.Spec.Broker = brokerName
+	}
+}
+
+// WithSubscriberRefForTrigger returns an option that adds a Subscriber Ref for the given Trigger.
+func WithSubscriberRefForTrigger(name string) TriggerOption {
+	return func(t *eventingv1alpha1.Trigger) {
+		if name != "" {
+			t.Spec.Subscriber = &eventingv1alpha1.SubscriberSpec{
+				Ref: pkgTest.CoreV1ObjectReference("Service", "v1", name),
```

Comment:
did you have a constants file for the kinds? can you add this one there?

Reply:
Added it there.